src/circuit/gadget/utilities.rs

```diff
@@ -0,0 +1,82 @@
+use halo2::{
+    circuit::{Cell, Chip, Layouter, Region},
+    plonk::{Advice, Column, Error, Permutation},
+};
+use pasta_curves::arithmetic::FieldExt;
+
+mod cond_swap;
+mod enable_flag;
+mod plonk;
+
+/// A variable representing a number.
+#[derive(Copy, Clone, Debug)]
+pub struct CellValue<F: FieldExt> {
+    cell: Cell,
+    value: Option<F>,
+}
+
+pub trait Var<F: FieldExt> {
+    fn new(cell: Cell, value: Option<F>) -> Self;
+    fn cell(&self) -> Cell;
+    fn value(&self) -> Option<F>;
+}
+
+impl<F: FieldExt> Var<F> for CellValue<F> {
+    fn new(cell: Cell, value: Option<F>) -> Self {
+        Self { cell, value }
+    }
+
+    fn cell(&self) -> Cell {
+        self.cell
+    }
+
+    fn value(&self) -> Option<F> {
+        self.value
+    }
+}
+
+pub trait UtilitiesInstructions<F: FieldExt>: Chip<F> {
+    type Var: Var<F>;
+
+    fn load_private(
+        &self,
+        mut layouter: impl Layouter<F>,
+        column: Column<Advice>,
+        value: Option<F>,
+    ) -> Result<Self::Var, Error> {
+        layouter.assign_region(
+            || "load private",
```

Comment:
Non-blocking: I'd make this annotation a parameter.

Reply:
This method takes a `mut layouter: impl Layouter<F>` which forces the caller to namespace the layouter.